src/routes/ArticleDetailPage.jsx

```diff
@@ -1,7 +1,51 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import ArticleFinder from "../apis/ArticleFinder";
 
 const ArticleDetailPage = () => {
-  return <div>Article Detail Page</div>;
+  const { id } = useParams();
+  const [selectedArticle, setSelectedArticle] = useState(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await ArticleFinder.get(`/articles/${id}`);
+        setSelectedArticle(response.data.article);
+      } catch (err) {
+        console.error(err);
+      }
+    };
+    fetchData();
+  }, [id]);
```

Comment:
Worth noting that using navigate here is somewhat overkill, as the<Link> tag e.g. `<Link to="/about">About</Link>` achieves the exact same thing while being less code overall. We usually use navigate when you need more control over when the navigation happens, but that's not really the case here